### PR TITLE
[zh-cn] translate for OfflineAudioContext

### PR DESCRIPTION
**Pull request by fuchunhui**

### CODE REVIEW

File: files/zh-cn/web/api/offlineaudiocontext/index.md [hunk 1]
```diff
@@ -1,106 +1,115 @@
...
-// 使用 XHR 去加载一个音轨，
-// 使用 decodeAudioData 去解码，
-// 使用 OfflineAudioContext 去渲染它
+## 示例
 
-function getData() {
-  request = new XMLHttpRequest();
+### 使用离线音频上下文播放音频
+
+这个例子中，我们声明了 {{domxref("AudioContext")}} 和 `OfflineAudioContext` 对象。我们使用 `AudioContext` 加载一个音频块 {{domxref("fetch()")}}，然后使用 `OfflineAudioContext` 去渲染音频并得到一个 into an {{domxref("AudioBufferSourceNode")}}，并播放这个音轨。在离线音频处理图建立后，我们使用 {{domxref("OfflineAudioContext.startRendering")}} 来渲染它成为 {{domxref("AudioBuffer")}}。
```
Comment (tianyeeT) [inline files/zh-cn/web/api/offlineaudiocontext/index.md:57]:
```suggestion
这个例子中，我们声明了 {{domxref("AudioContext")}} 和 `OfflineAudioContext` 对象。我们使用 `AudioContext` 加载一个音频块 {{domxref("fetch()")}}，然后使用 `OfflineAudioContext` 去渲染音频并得到一个 {{domxref("AudioBufferSourceNode")}}，并播放这个音轨。在离线音频处理图建立后，我们使用 {{domxref("OfflineAudioContext.startRendering")}} 来渲染它成为 {{domxref("AudioBuffer")}}。
```

File: files/zh-cn/web/api/offlineaudiocontext/index.md [hunk 2]
```diff
@@ -1,106 +1,115 @@
...
-> **备注：** 为了获取可以运行的例子，请看我们在 Github 的仓库 [offline-audio-context-promise](https://mdn.github.io/webaudio-examples/offline-audio-context-promise/) （也可以看到 [源代码](https://github.com/mdn/webaudio-examples/tree/master/offline-audio-context-promise)。）
+通过使用 [`addEventListener()`](/zh-CN/docs/Web/API/EventTarget/addEventListener) 或给这个接口的 `oneventname` 属性指定一个事件监听器来监听这些事件。
 
-```js
-// 定义一个在线或者离线的音频上下文
+- [`complete`](/zh-CN/docs/Web/API/OfflineAudioContext/complete_event)
 
-var audioCtx = new AudioContext();
-var offlineCtx = new OfflineAudioContext(2, 44100 * 40, 44100);
+  - : Fired when the rendering of an offline audio context is complete.
```
Comment (tianyeeT) [inline files/zh-cn/web/api/offlineaudiocontext/index.md:48]:
```suggestion
  - : 当离线音频上下文的渲染完成时触发。
```

File: files/zh-cn/web/api/offlineaudiocontext/index.md [hunk 3]
```diff
@@ -1,106 +1,115 @@
...
+
+这个例子中，我们声明了 {{domxref("AudioContext")}} 和 `OfflineAudioContext` 对象。我们使用 `AudioContext` 加载一个音频块 {{domxref("fetch()")}}，然后使用 `OfflineAudioContext` 去渲染音频并得到一个 into an {{domxref("AudioBufferSourceNode")}}，并播放这个音轨。在离线音频处理图建立后，我们使用 {{domxref("OfflineAudioContext.startRendering")}} 来渲染它成为 {{domxref("AudioBuffer")}}。
+
+当 `startRendering()` 的 `Promise` 解决后，渲染也完成了，在 `Promise` 内可以获得输出的 `AudioBuffer`。
 
-  request.open("GET", "viper.ogg", true);
+在此刻，我们创建了一个另外的音频上下文，在它里面创建了一个 {{domxref("AudioBufferSourceNode")}}，并且设置它的 buffer 为之前生成的 Promise 中的 `AudioBuffer`，这样它就可以作为简单标准音频图来播放了。
 
-  request.responseType = "arraybuffer";
+> **备注：** 你可以 [在线运行完整示例](https://mdn.github.io/webaudio-examples/offline-audio-context-promise/)，或 [查看源代码](https://github.com/mdn/webaudio-examples/blob/main/offline-audio-context-promise/)
```
Comment (tianyeeT) [inline files/zh-cn/web/api/offlineaudiocontext/index.md:63]:
```suggestion
> **备注：** 你可以[在线运行完整示例](https://mdn.github.io/webaudio-examples/offline-audio-context-promise/)，或[查看源代码](https://github.com/mdn/webaudio-examples/blob/main/offline-audio-context-promise/)。
```

File: files/zh-cn/web/api/offlineaudiocontext/index.md [hunk 4]
```diff
@@ -1,106 +1,115 @@
...
 
-## 示例
+### 废弃的方法
 
-这个简单的例子中，我们声明了 {{domxref("AudioContext")}} 和 `OfflineAudioContext` 对象。我们使用 `AudioContext` 去加载一个 XHR（{{domxref("AudioContext.decodeAudioData")}}）获取的音轨，然后使用 `OfflineAudioContext` 去渲染音频并得到一个 into an {{domxref("AudioBufferSourceNode")}}，并播放这个音轨。在离线音频处理图建立后，你需要去使用 {{domxref("OfflineAudioContext.startRendering")}} 来渲染它成为 {{domxref("AudioBuffer")}}。
+- {{domxref("OfflineAudioContext.resume()")}}
+  - : 恢复一个被暂停的音频的时间进程。
 
-当 `startRendering()` 的 `Promise` 解决后，渲染也完成了，在 `Promise` 内可以获得输出的 `AudioBuffer`。
+> **备注：** `resume()` 方法仍然是可用的 — 现在它定义在 {{domxref("BaseAudioContext")}} 接口中（参见 {{domxref("AudioContext.resume")}}），通过{{domxref("AudioContext")}} 接口和 {{domxref("OfflineAudioContext")}} 接口都可以被访问。
```
Comment (tianyeeT) [inline files/zh-cn/web/api/offlineaudiocontext/index.md:40]:
```suggestion
> **备注：** `resume()` 方法仍然是可用的——现在它定义在 {{domxref("BaseAudioContext")}} 接口中（参见 {{domxref("AudioContext.resume")}}），通过{{domxref("AudioContext")}} 接口和 {{domxref("OfflineAudioContext")}} 接口都可以被访问。
```
或者改成句号

File: files/zh-cn/web/api/offlineaudiocontext/index.md [hunk 5]
```diff
@@ -1,106 +1,115 @@
...
 
-- {{domxref("OfflineAudioContext.OfflineAudioContext()")}}
+- {{domxref("OfflineAudioContext.OfflineAudioContext()", "OfflineAudioContext()")}}
   - : 创建一个新的 `OfflineAudioContext` 实例。
 
-## 属性
+## 实例属性
 
-_从父级 {{domxref("AudioContext")}} 获取属性。_
+_从父级 {{domxref("BaseAudioContext")}} 获取属性。_
```
Comment (yin1999) [inline files/zh-cn/web/api/offlineaudiocontext/index.md:21]:
```suggestion
_从其父接口 {{domxref("BaseAudioContext")}} 继承属性。_
```

File: files/zh-cn/web/api/offlineaudiocontext/index.md [hunk 6]
```diff
@@ -1,106 +1,115 @@
...
 
-- {{domxref("OfflineAudioContext.resume()")}}
-  - : 恢复一个被暂停的音频的时间进程。
 - {{domxref("OfflineAudioContext.suspend()")}}
   - : 在指定的时间安排音频暂停时间进程，并且通过 Promise 返回。
 - {{domxref("OfflineAudioContext.startRendering()")}}
   - : 开始渲染音频，考虑当前连接和当前计划的修改。这个页面涵盖基于事件的和基于 Promise 的版本。
 
-## 示例
+### 废弃的方法
```
Comment (yin1999) [inline files/zh-cn/web/api/offlineaudiocontext/index.md:35]:
```suggestion
### 已弃用的方法
```

File: files/zh-cn/web/api/offlineaudiocontext/index.md [hunk 7]
```diff
@@ -1,106 +1,115 @@
...
-// 定义一个在线或者离线的音频上下文
+- [`complete`](/zh-CN/docs/Web/API/OfflineAudioContext/complete_event)
 
-var audioCtx = new AudioContext();
-var offlineCtx = new OfflineAudioContext(2, 44100 * 40, 44100);
+  - : 当离线音频上下文的渲染完成时触发。
 
-source = offlineCtx.createBufferSource();
+- [`complete`](/zh-CN/docs/Web/API/OfflineAudioContext/complete_event)
+  - : 当离线的音频进程完成渲染时触发。
```
Comment (yin1999) [inline files/zh-cn/web/api/offlineaudiocontext/index.md:51]:
```suggestion
```

File: files/zh-cn/web/api/offlineaudiocontext/index.md [hunk 8]
```diff
@@ -1,106 +1,115 @@
...
```
Comment (yin1999) [inline files/zh-cn/web/api/offlineaudiocontext/index.md:52]:
```suggestion
```

File: files/zh-cn/web/api/offlineaudiocontext/index.md [hunk 9]
```diff
@@ -1,106 +1,115 @@
...
-// 使用 XHR 去加载一个音轨，
-// 使用 decodeAudioData 去解码，
-// 使用 OfflineAudioContext 去渲染它
+## 示例
 
-function getData() {
-  request = new XMLHttpRequest();
+### 使用离线音频上下文播放音频
+
+这个例子中，我们声明了 {{domxref("AudioContext")}} 和 `OfflineAudioContext` 对象。我们使用 `AudioContext` 加载一个音频块 {{domxref("fetch()")}}，然后使用 `OfflineAudioContext` 去渲染音频并得到一个 {{domxref("AudioBufferSourceNode")}}，并播放这个音轨。在离线音频处理图建立后，我们使用 {{domxref("OfflineAudioContext.startRendering")}} 来渲染它成为 {{domxref("AudioBuffer")}}。
```
Comment (yin1999) [inline files/zh-cn/web/api/offlineaudiocontext/index.md:57]:
```suggestion
这个例子中，我们声明了 {{domxref("AudioContext")}} 和 `OfflineAudioContext` 对象。我们使用 `AudioContext` 加载一个音轨的 {{domxref("fetch()")}}，然后使用 `OfflineAudioContext` 去渲染音频并得到一个 {{domxref("AudioBufferSourceNode")}}，并播放这个音轨。在离线音频处理图建立后，我们使用 {{domxref("OfflineAudioContext.startRendering")}} 来将它渲染为 {{domxref("AudioBuffer")}}。
```

File: files/zh-cn/web/api/offlineaudiocontext/index.md [hunk 10]
```diff
@@ -1,106 +1,115 @@
...
+
+当 `startRendering()` 的 `Promise` 解决后，渲染也完成了，在 `Promise` 内可以获得输出的 `AudioBuffer`。
```
Comment (yin1999) [inline files/zh-cn/web/api/offlineaudiocontext/index.md:59]:
```suggestion
当 `startRendering()` 的 promise 兑现后，渲染也完成了，promise 会返回输出的 `AudioBuffer`。
```

**Review by yin1999 — COMMENTED** [review]
一些建议，可以在 [Fires changed](https://github.com/mdn/translated-content/pull/21611/files) 标签页[一次性应用建议](https://docs.github.com/zh/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request)。

**Review by yin1999 — APPROVED** [review]
Thank you @fuchunhui

**Pull request merged**